include/seqan/seeds/seeds_global_chaining.h

```diff
@@ -153,98 +177,108 @@ chainSeedsGlobally(
+    // We build a list of "intermediate solutions" (referred to as *L* in the Gusfield book).
+    // Each such solution is represented by the triple (end position in dim1,
     // value of best chain so far, last seed of the chain).
     typedef Triple<TPosition, TSize, unsigned> TIntermediateSolution;
     typedef std::multiset<TIntermediateSolution> TIntermediateSolutions;
     typedef typename TIntermediateSolutions::iterator TIntermediateSolutionsIterator;
 
     // For all interval points...
     TIntermediateSolutions intermediateSolutions;
-    for (TIntervalPointsIterator it = begin(intervalPoints), itEnd = end(intervalPoints); it != itEnd; ++it) {
+    for (TIntervalPointsIterator it_k = begin(intervalPoints), itEnd = end(intervalPoints); it_k != itEnd; ++it_k) {
         // The seed belonging ot the interval point is seed k.
-        TSeed const & seedK = seeds[it->i3];
+        TSeed const & seed_k = seeds[it_k->i3];
 
         // std::cout << "Processing interval point (" << it->i1 << ", " << it->i2 << ", " << it->i3 << ")" << std::endl;
-        if (it->i2) {  // Is is begin point.
-            // Find the closest seed (in dimension 1) to seed k with an
-            // entry in intermediateSolutions whose end coordinate in
-            // dimension 1 is <= the begin coordinate in dimension 1
-            // of seedK.
+        if (it_k->i2) {  // Is is begin point.
+            // Find the closest seed j (in y-dimension) with an
+            // entry in L whose end coordinate is less or equal the begin position of k.
             //
             // STL gives us upper_bound which returns a pointer to the
             // *first* one that compares greater than the reference
             // one.  Searching for the this one and decrementing the
             // result iterator gives the desired result.
-            TIntermediateSolution referenceSolution(beginPositionV(seedK), std::numeric_limits<TSize>::max(), std::numeric_limits<unsigned>::max());
-            // std::cout << "    intermediateSolutions.upper_bound(" << beginPositionV(seedK) << ")" << std::endl;
-            TIntermediateSolutionsIterator itJ = intermediateSolutions.upper_bound(referenceSolution);
-            if (itJ == intermediateSolutions.begin()) {
-                if (intermediateSolutions.size() > 0 &&
-                    intermediateSolutions.rbegin()->i1 <= beginPositionV(seedK)) {
-                    itJ = intermediateSolutions.end();
-                    --itJ;
-                } else {
-                    continue;
-                }
-            } else {
-                SEQAN_ASSERT_GT(intermediateSolutions.size(), 0u);  // TODO(holtgrew): Remove this assertion?
-                --itJ;
+            TIntermediateSolution referenceSolution(beginPositionV(seed_k), std::numeric_limits<TSize>::max(), std::numeric_limits<unsigned>::max());
+            // std::cout << "    intermediateSolutions.upper_bound(" << beginPositionV(seed_k) << ")" << std::endl;
```

Comment:
remove debug or mark clearly as such, e.g. by
```cpp
#if 0 // DEBUG functionality
    std::cout << "    intermediateSolutions.upper_bound(" << beginPositionV(seed_k) << ")" << std::endl;
#endif
```

Reply:
This is untouched code. Just moved around by git a little.